protobuf/types/struct.go

```diff
@@ -0,0 +1,38 @@
+package types
+
+import (
+	"sort"
+)
+
+// KeyValue is a simple key/value representation of one field of a Struct.
+type KeyValue struct {
+	Key   string
+	Value *Value
+}
+
+// MarshalAmino transforms the Struct to an array of key/value.
+func (m Struct) MarshalAmino() ([]KeyValue, error) {
+	p := make([]KeyValue, 0)
+	fieldKeys := make([]string, 0)
```

Comment:
Use len(m.Fields) for preallocating space for sliceses. Then used index to insert instead of append